logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/message/MessageSender.kt

```diff
@@ -97,21 +101,28 @@ class MessageSenderImpl(
     }
 
     override suspend fun sendMessage(message: Message.Regular): Either<CoreFailure, Unit> = attemptToSend(message)
-        .flatMap { messageRemoteTime ->
-            messageRepository.updateMessageDate(message.conversationId, message.id, messageRemoteTime)
-                .map { messageRemoteTime }
-        }
-        .flatMap { messageRemoteTime ->
-            messageRepository.updateMessageStatus(MessageEntity.Status.SENT, message.conversationId, message.id)
-                .map { messageRemoteTime }
-        }
-        .flatMap { messageRemoteTime ->
-            // this should make sure that pending messages are ordered correctly after one of them is sent
-            messageRepository.updatePendingMessagesAddMillisToDate(
-                message.conversationId,
-                timeParser.calculateMillisDifference(message.date, messageRemoteTime)
-            )
-        }
+        .onSuccess { messageRemoteTime ->
+            updateDatesOfMessagesWithServerTime(message, messageRemoteTime)
+        }.map { }
```

Comment:
Maybe we can simplify that and get rid of empty `map` by doing it like this:
```
attemptToSend(message).map { messageRemoteTime ->
    updateDatesOfMessagesWithServerTime(message, messageRemoteTime)
}
```
because `updateDatesOfMessagesWithServerTime` returns `Unit` 
and `Either<L, R>.map` executes its lambda only if there's a "right" value emitted

Reply:
Oh yeah indeed! Done it.